install/playbooks/roles/dovecot/files/access-log.sh

```diff
@@ -107,9 +108,31 @@ fi
 # Remove the new lines from the details before storing them in the database
 smallDetails=$(echo "$DETAILS" | tr '\n' ';' | sed -r 's/(^;|;$)//g')
 
+# Get the Internet Service Provider from http://ip-api.com/
+# The limit is 150 requests per minute from an IP address.
+# This is particularly useful if you are using only ISP,
+# regardless of the IP address
+isp='unknown'
+
+if [ "$isPrivate" = "1" ]; then
+    isp='private'
+elif [ "$ALLOW_EXTERNAL_QUERIES" = "YES" ]; then
+    # set curl options
+    # -s: silent
+    # -f: fail silently, return an empty string on failure
+    # -m 10: wait maximum 10 seconds for the whole process
+    options='-s -f -m 10'
+    url="http://ip-api.com/line/$IP?fields=isp"
+    isp=$(curl $options "$url" | sed "s/'/\\'/g")
```

Comment:
The single quote escape for the SQL query should be done with two single quotes:
https://stackoverflow.com/questions/603572/escape-single-quote-character-for-use-in-an-sqlite-query/603579#603579

I am not sure some backslash escapes are needed for the shell script variables and strings. It looks ok to me.